lib/i18n/request.ts

```diff
@@ -0,0 +1,68 @@
+import accepts from 'accepts';
+import cookie from 'cookie';
+import { pick } from 'lodash';
+import { NextPageContext } from 'next';
+
+const supportedLanguages = ['en'];
+
+const languages = require.context('../../lang', false, /\.json$/i);
+
+languages.keys().forEach(element => {
+  const match = element.match(/\.?\/?([^.]+)\.json$/);
+  if (match) {
+    supportedLanguages.push(match[1]);
+  }
+});
+
+type IntlProps = {
+  language?: string;
+  locale: string;
+  messages: Record<string, string>;
+};
+
+export function getIntlProps(ctx: NextPageContext): IntlProps {
+  if (!ctx?.req) {
+    return pick(window.__NEXT_DATA__.props, 'language', 'locale', 'messages');
+  }
+
+  const { language, locale } = getRequestIntl(ctx.req);
+  const messages = getLocaleMessages(locale);
+  return {
+    language,
+    locale,
+    messages,
+  };
+}
+
+export function getRequestIntl(req: NextPageContext['req']): Pick<IntlProps, 'language' | 'locale'> {
+  let language: string, locale: string;
+
+  const url = new URL(req.url, `http://${req.headers.host}`);
+  const queryLanguage = url.searchParams.get('language');
```

Comment:
I don't get this bit: isn't the host supposed to contain only the domain, e.g. `opencollective.com`, without any URL parameters?

Reply:
`req.url` contains the request path + params 

js docs:
```
Request URL string. This contains only the URL that is present in the actual HTTP request. Take the following request:

GET /status?name=ryan HTTP/1.1
Accept: text/plain
To parse the URL into its parts:

new URL(request.url, `http://${request.headers.host}`);
When request.url is '/status?name=ryan' and request.headers.host is'localhost:3000':

$ node
> new URL(request.url, `http://${request.headers.host}`)
URL {
  href: 'http://localhost:3000/status?name=ryan',
  origin: 'http://localhost:3000',
  protocol: 'http:',
  username: '',
  password: '',
  host: 'localhost:3000',
  hostname: 'localhost',
  port: '3000',
  pathname: '/status',
  search: '?name=ryan',
  searchParams: URLSearchParams { 'name' => 'ryan' },
  hash: ''
}
```

Reply:
Interesting, thanks for sharing